worlds/mmx/__init__.py

```diff
@@ -0,0 +1,402 @@
+        slot_data["hadouken_in_pool"] = self.options.hadouken_in_pool.value
+        slot_data["pickupsanity"] = self.options.pickupsanity.value
+        slot_data["logic_boss_weakness"] = self.options.logic_boss_weakness.value
+        slot_data["logic_leg_sigma"] = self.options.logic_leg_sigma.value
+        slot_data["logic_charged_shotgun_ice"] = self.options.logic_charged_shotgun_ice.value
+        slot_data["sigma_all_levels"] = self.options.sigma_all_levels.value
+        value = 0
+        sigma_open = self.options.sigma_open.value
+        if "Medals" in sigma_open:
+            value |= 0x01
+        if "Weapons" in sigma_open:
+            value |= 0x02
+        if "Armor Upgrades" in sigma_open:
+            value |= 0x04
+        if "Heart Tanks" in sigma_open:
+            value |= 0x08
+        if "Sub Tanks" in sigma_open:
+            value |= 0x10
+        slot_data["sigma_open"] = value
+        slot_data["sigma_medal_count"] = self.options.sigma_medal_count.value
+        slot_data["sigma_weapon_count"] = self.options.sigma_weapon_count.value
+        slot_data["sigma_upgrade_count"] = self.options.sigma_upgrade_count.value
+        slot_data["sigma_heart_tank_count"] = self.options.sigma_heart_tank_count.value
+        slot_data["sigma_sub_tank_count"] = self.options.sigma_sub_tank_count.value
+
+        # Write boss weaknesses to slot_data (and for UT)
+        slot_data["boss_weaknesses"] = {}
+        slot_data["weakness_rules"] = {}
+        for boss, entries in self.boss_weaknesses.items():
+            slot_data["weakness_rules"][boss] = entries.copy()
+            slot_data["boss_weaknesses"][boss] = []
+            for entry in entries:
+                slot_data["boss_weaknesses"][boss].append(entry[1])
+                
+        return slot_data
+
+
+    def generate_early(self) -> None:
+        if ItemName.legs not in self.options.start_inventory_from_pool and self.options.early_legs:
+            self.multiworld.early_items[self.player][ItemName.legs] = 1
+            
+        # Generate weaknesses
+        self.boss_weakness_data = {}
+        self.boss_weaknesses = {}
+        handle_weaknesses(self)
+
+
+    def interpret_slot_data(self, slot_data: dict) -> Dict[str, Any]:
+        local_weaknesses = dict()
+        for boss, entries in slot_data["weakness_rules"].items():
+            local_weaknesses[boss] = entries.copy()
+        return {"weakness_rules": local_weaknesses}
+    
+
+    def write_spoiler(self, spoiler_handle: TextIO) -> None:
+        if self.options.boss_weakness_rando != "vanilla":
+            spoiler_handle.write(f"\nMega Man X boss weaknesses for {self.multiworld.player_name[self.player]}:\n")
+            
+            for boss, data in self.boss_weaknesses.items():
+                weaknesses = ""
+                for i in range(len(data)):
+                    weaknesses += f"{weapon_id[data[i][1]]}, "
+                weaknesses = weaknesses[:-2]
```

Comment:
```suggestion
                weaknesses = ", ".join([weapon_id[weak[1]] for weak in data])
```
I think this will give the same result. You should test to make sure.